source/MaterialXGenShader/ShaderGenerator.h

```diff
@@ -185,7 +185,8 @@ class ShaderGenerator
     /// @param qualifier Optional qualifier to add before the variable declaration.
     /// Qualifiers are specified by the syntax for the generator.
     /// @param shader Shader to emit to.
-    virtual void emitVariableBlock(const Shader::VariableBlock& block, const string& qualifier, Shader& shader);
+    /// @param separator Separator to use between variables.
+    virtual void emitVariableBlock(const Shader::VariableBlock& block, const string& qualifier, Shader& shader, const string& separator);
```

Comment:
Nitpick: for consistency could you flip the last two arguments so the 'shader' output argument is given last.